.github/workflows/python-package.yml

```diff
@@ -42,15 +42,40 @@ jobs:
         python -m pip install --upgrade pip
         python -m pip install flake8 pytest
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Test with pytest
+      run: pytest --benchmark-skip
     - name: Lint with flake8
       run: |
         # stop the build if there are Python syntax errors or undefined names
         #flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         #flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with pytest
+
+  cb:
+    # Benchmark only after the correctness has been tested by CI,
+    # and then run benchmark only once (sampling with only one Python version).
+    needs: ci
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Install dependencies
       run: |
-        pytest
+        python -m pip install --upgrade pip
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Run benchmark
+      run: pytest --benchmark-only --benchmark-json benchmark.json --log-cli-level INFO tests/test_benchmark.py
+    - name: Render benchmark result
+      uses: benchmark-action/github-action-benchmark@v1
+      with:
+        tool: 'pytest'
```

Comment:
Should we set alert-threshold? By default it's 200% (so twice as slow perf regression). In .NET we set it to 130%

Reply:
Based on the [diagram so far](https://azuread.github.io/microsoft-authentication-library-for-python/dev/bench/), the peak and valley can easily have a swing of 160% while the underlying test subject did not change. Since each sample is individually reliable (they have small stdev), it seems like the fluctuation was caused by the Github Action agent?

Anyway, 130% would already result in false positive for MSAL Python, we will stay with the default 200% for now. (At one point, we even discussed using a much higher alert-threshold, didn't we?)